tests/system/feature_store/spark_hadoop_test_base.py

```diff
@@ -0,0 +1,154 @@
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+from enum import Enum
+
+import pandas as pd
+import pytest
+
+import mlrun.feature_store as fstore
+from mlrun import store_manager
+from mlrun.datastore.sources import ParquetSource
+from mlrun.datastore.targets import (
+    ParquetTarget,
+)
+from tests.system.base import TestMLRunSystem
+from tests.system.feature_store.expected_stats import expected_stats
+
+
+class Deployment(Enum):
+    Local = 1
+    Remote = 2
+
+
+@pytest.mark.enterprise
+class SparkHadoopTestBase(TestMLRunSystem):
+    """
+    This class serves as a support tool for testing Spark with file systems based on Hadoop,
+    both as a source and a target. It is not designed for testing Spark operators.
+
+    To conduct a test with a specific filesystem, you should derive your test class from `SparkHadoopTestBase`.
+    In your test class, override the `custom_setup_class()` method following the example provided below.
+    Then, in your test, ensure to call the `do_test()` method of this class.
+
+    Here's an illustration of how a derived test class might look:
+    class TestFeatureStoreSparkEngine(SparkHadoopTestBase):
+        @classmethod
+        def custom_setup_class(cls):
+            cls.configure_namespace("mytest")
+            cls.configure_image_deployment(
+                Deployment.Remote,
+                "azure-storage-blob git+https://github.com/mlrun/mlrun.git@development",
+            )
+        def test_basic_remote_spark_ingest_ds_basic(self):
+            bucket = 'my_bucket'
+            ds_profile = DatastoreProfileBasic(name=self.ds_profile_name)
+            register_temporary_client_datastore_profile(ds_profile)
+            self.project.register_datastore_profile(ds_profile)
+            self.ds_upload_src(ds_profile, bucket)
+            self.do_test(
+                self.ds_src_path(ds_profile, bucket),
+                self.ds_target_path(ds_profile, bucket),
+            )
+
+    It is also possible to run the tests locally if you have pyspark installed. To run locally, call
+    configure_image_deployment(deployment_type=Deployment.Local). This can be very useful for debugging.
+    """
+
+    @classmethod
+    def configure_namespace(cls, test_suite_name="hadoop"):
+        cls.project_name = "fs-system-spark-engine-" + test_suite_name
+        cls.ds_profile_name = "ds_profile_spark_" + test_suite_name
+        cls.remote_function_name = (
+            "remote-spark-default-image-deploy-temp-" + test_suite_name
+        )
+        cls.spark_service_name = "dummy-spark-" + test_suite_name
```

Comment:
I don't think the service name should change between suites... The suite doesn't create the service – it depends on the system to already have it running. We don't want to have to create a service per suite.

Actually, this should be the same as in `TestFeatureStoreSparkEngine`, getting it from `env.yml`:
```python
    @classmethod
    def _init_env_from_file(cls):
        env = cls._get_env_from_file()
        if cls.run_local:
            cls.spark_service = None
        else:
            cls.spark_service = env["MLRUN_SYSTEM_TESTS_DEFAULT_SPARK_SERVICE"]
```